Do not accept empty string values from the lsp settings

tgodzik opened the pull request.

olafurpg commented on metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala:
```diff
@@ -133,11 +133,16 @@ object UserConfiguration {
+    def getKey[A](
+        key: String,
+        f: JsonElement => Option[A],
+        isEmpty: JsonElement => Boolean = _ => false
```
Instead of passing in `isEmpty = ...`, the `f` method can implement the filter instead.

tgodzik replied:
Hmm... that will not work since the f method is invoked after we get the value.

olafurpg commented on metals/src/main/scala/scala/meta/internal/metals/UserConfiguration.scala:
```diff
@@ -155,8 +160,12 @@ object UserConfiguration {
               None
             }, Some(_))
```
```suggestion
            }, Some(_).filter(_.nonEmpty))
```

olafurpg commented:
There is BTW a `UserConfigurationSuite`, we should add a test case for this

tgodzik commented:
> There is BTW a `UserConfigurationSuite`, we should add a test case for this

Added test cases and switched around the logic to make this simpler. First properties and the settings.
